ansible/collections/ansible_collections/nhsd/apigee/plugins/action/deploy_product.py

```diff
@@ -80,5 +90,9 @@ def run(self, tmp=None, task_vars=None):
             return new_response
 
         result["product"] = new_response["response"]["body"]
-        result["changed"] = True
+        if diff_mode:
+            result["diff"][0]["after"]: {
+                k: v for k, v in result["product"].items() if k not in keys_to_ignore
```

Comment:
This pattern keeps cropping up a lot, it's worth turning this into a function e.g. `exclude(result["product"], keys_to_ignore)`